sdk/identity/azure-identity/src/workload_identity_credential.cpp

```diff
@@ -96,6 +109,17 @@ WorkloadIdentityCredential::WorkloadIdentityCredential(
               "urn%3Aietf%3Aparams%3Aoauth%3Aclient-assertion-type%3Ajwt-bearer" // cspell:disable-line
               "&client_id=")
         + Url::Encode(clientId);
+
+    IdentityLog::Write(
+        IdentityLog::Level::Informational, GetCredentialName() + " was created successfully.");
+  }
+  else
+  {
+    IdentityLog::Write(
```

Comment:
Ideally, both constructors would have the common code in one place, so that not only there will be less code repetition, as here, as well as for most of the things you would only need to update the logic in one place.

If constructor delegation does not work, maybe we can have `private: Init(...args...)` which would be called from both constructors.
Or, there could be a 3rd private constructor, to which both other constructors do delegate with different parameters.

You don't have to fix in this PR; if you agree to look into it later, please log a work item.

Reply:
We discussed this previously (same concern about other credential implementations), where the delegate pattern to avoid slight code duplication isn't the best.

WIC used to have the delegate ctor pattern, but I changed it (see previous discussion - https://github.com/Azure/azure-sdk-for-cpp/pull/4872#discussion_r1291853951).

I can revisit this separately:
https://github.com/Azure/azure-sdk-for-cpp/issues/4874